testsrc/org/mozilla/javascript/tests/commonjs/module/testCustomGlobal.js

```diff
@@ -0,0 +1,24 @@
+var assert = require("assert");
+
+assert.strictEqual(test(1), 2);
+
+var testWrap = function(arg) { return test(arg); };
+
+assert.strictEqual(testWrap(2), 3);
+
+var testWrap2 = function() { return function(arg) { return test(arg); }; };
+
+assert.strictEqual(testWrap2()(3), 4);
+
+require('testCustomGlobal2');
+
+var throws = false;
```

Comment:
I'm not sure if this helps, but did you try using "assertThrows" here? It ensures that a function throws an exception and it lets you check the assertion's type and message.

Reply:
I tried using `assert["throws"]`, and it works, but it only checks the assertion's type. The message parameter is used as the error message if the error is not thrown. I implemented it this way so that I could check the assertion's message.

Reply:
You're right -- assertThrows has a third argument that's purportedly trying to test the message, but it seems to predate all modern JavaScript and doesn't actually do that. I think that we should fix it to check the "message" property of the error object.

But I don't think it's worth holding up this PR for that reason, and otherwise this looks good. Thanks!